src/waveform/renderers/allshader/waveformrendererrgb.cpp

```diff
@@ -123,79 +129,103 @@ void WaveformRendererRGB::paintGL() {
-                u8maxHigh = math_max(u8maxHigh, waveformData.filtered.high);
+                u8maxLow[signalChn] = math_max(u8maxLow[signalChn], waveformData.filtered.low);
+                u8maxMid[signalChn] = math_max(u8maxMid[signalChn], waveformData.filtered.mid);
+                u8maxHigh[signalChn] = math_max(u8maxHigh[signalChn], waveformData.filtered.high);
                 u8maxAllChn[chn] = math_max(u8maxAllChn[chn], waveformData.filtered.all);
             }
         }
-
-        // Cast to float
-        float maxLow = static_cast<float>(u8maxLow);
-        float maxMid = static_cast<float>(u8maxMid);
-        float maxHigh = static_cast<float>(u8maxHigh);
         float maxAllChn[2]{static_cast<float>(u8maxAllChn[0]), static_cast<float>(u8maxAllChn[1])};
-        // Uncomment to undo scaling with pow(value, 2.0f * 0.316f) done in analyzerwaveform.h
-        // float maxAllChn[2]{unscale(u8maxAllChn[0]), unscale(u8maxAllChn[1])};
-
-        // Calculate the squared magnitude of the maxLow, maxMid and maxHigh values.
-        // We take the square root to get the magnitude below.
-        const float sum = math_pow2(maxLow) + math_pow2(maxMid) + math_pow2(maxHigh);
-
-        // Apply the gains
-        maxLow *= lowGain;
-        maxMid *= midGain;
-        maxHigh *= highGain;
-
-        // Calculate the squared magnitude of the gained maxLow, maxMid and maxHigh values
-        // We take the square root to get the magnitude below.
-        const float sumGained = math_pow2(maxLow) + math_pow2(maxMid) + math_pow2(maxHigh);
-
-        // The maxAll values will be used to draw the amplitude. We scale them according to
-        // magnitude of the gained maxLow, maxMid and maxHigh values
-        if (sum != 0.f) {
-            // magnitude = sqrt(sum) and magnitudeGained = sqrt(sumGained), and
-            // factor = magnitudeGained / magnitude, but we can do with a single sqrt:
-            const float factor = std::sqrt(sumGained / sum);
-            maxAllChn[0] *= factor;
-            maxAllChn[1] *= factor;
-        }
 
-        // Use the gained maxLow, maxMid and maxHigh values to calculate the color components
-        float red = maxLow * low_r + maxMid * mid_r + maxHigh * high_r;
-        float green = maxLow * low_g + maxMid * mid_g + maxHigh * high_g;
-        float blue = maxLow * low_b + maxMid * mid_b + maxHigh * high_b;
-
-        // Normalize the color components using the maximum of the three
-        const float maxComponent = math_max3(red, green, blue);
-        if (maxComponent == 0.f) {
-            // Avoid division by 0
-            red = 0.f;
-            green = 0.f;
-            blue = 0.f;
-        } else {
-            const float normFactor = 1.f / maxComponent;
-            red *= normFactor;
-            green *= normFactor;
-            blue *= normFactor;
-        }
+        for (int chn = 0;
+                // In case we don't render individual color per channel, all the
```

Comment:
putting this comment inside the for statement doesn't help readablilty. I propose moving it outside and using 
```
int nchns = (splitLeftRight && !m_isSlipRenderer ? 2 : 1;
for (int chn = 0; chn < nchns; chn++) {
```